lib/outputs/ElasticSearchOutput.js

```diff
@@ -2,111 +2,79 @@ var util = require('util'),
- * @param {object} [options.agent] An http or https agent, useful for providing your own CA, among other things
+ * @param {Object} [options.client]  Elasticsearch javascript client config.  See Elasticdocs for what can be specified here
+ * @param {Object} [options.options] Elasticsearch bulk request options see Elasticdocs for what can be specified here
+ *
  * @param {Number} [options.batchSize=100] The ideal size for each batch insertion request
  * @param {Number} [options.batchTimeout=1000] The time to wait for a batch to reach `options.batchSize`. If the batch size
  *      isn't met within this time the insertion request is sent with whatever is in the current batch
- * @param {Number} [options.port=9200] The port on the host to send requests to
- * @param {String} [options.typeField=] The top level field in the event#data object to use for the type value
- * @param {String} [options.defaultType='unknown'] The default type value if typeField is undefined
- * @param {String} [options.timestampField='timestamp'] The top level field in the event#data object to use for the timestamp value
  * @param {String} [options.name='ElasticSearch'] A name to use for logging, must be unique to other output plugins
  *
  * @constructor
  */
 var ElasticSearchOutput = function (options) {
-    ElasticSearchOutput.super_.call(this)
+    ElasticSearchOutput.super_.call(this);
 
     var self = this,
-        useOptions = options || {}
+        useOptions = options || {};
 
-    if (!useOptions.streamStash) {
-        throw new Error('The streamstash object was not provided to the constructor')
-    }
+    self.logger = useOptions.logger;
 
-    if (!useOptions.hostname) {
-        throw new Error('No hostname was provided')
+    var streamStash = useOptions.streamStash;
+    if (!streamStash) {
+        throw new Error('The streamstash object was not provided to the constructor')
     }
 
-    self.name = useOptions.name || 'ElasticSearch'
-    self.indexPrefix = useOptions.indexPrefix || 'logstash'
-
-    var streamStash = useOptions.streamStash
-
-    if (useOptions.tls) {
-        self.agent = useOptions.agent || new https.Agent({ maxSockets: 1000, rejectUnauthorized: true })
-        self.driver = https
-    } else {
-        self.agent = useOptions.agent || new http.Agent({ maxSockets: 1000 })
-        self.driver = http
-    }
+    self.name = useOptions.name || 'ElasticSearch';
 
-    self.httpOptions = {
-        hostname: useOptions.hostname,
-        port: useOptions.port || 9200,
-        path: '/_bulk',
-        method: 'POST',
-        agent: self.agent
-    }
+    self.elasticClient = new elasticsearch.Client(useOptions.client);
 
-    if (useOptions.auth) {
-        self.httpOptions.auth = useOptions.auth
-    }
+    self.options = useOptions.options  || {};
 
     self.queueOptions = {
         batchSize: useOptions.batchSize || 100,
         timeout: useOptions.batchTimeout || 1000
-    }
+    };
```

Comment:
; is entirely optional and should be avoided unless the situation specifically calls for it